### PR TITLE
Add Travis-CI setup

### LINKED ISSUE #19: Setup CI for testing changes
Was thinking that Travis could probably be setup to clone down the nodejs/node repo, and any other nodejs repos that are using this to see if changes to the config cause new issues.

### PR DESCRIPTION
**Pull request by nschonni**
Fixes #19

### CODE REVIEW

**Comment by nschonni** [comment]
Here is the CI status:
- first commit https://travis-ci.com/nschonni/remark-preset-lint-node/builds/126135817 had a bunch of first-heading level warnings
- second commit https://travis-ci.com/nschonni/remark-preset-lint-node/builds/126136224

Submitted https://github.com/nodejs/node/pull/29463 to fix the remaining error in the CI

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,6 @@ jspm_packages
 
 # Optional REPL history
 .node_repl_history
+
+# Cloned nodejs/node for testing
+test/
```
Comment (Trott) [inline .gitignore]: Nit-pick: Add a newline to be consistent with all the other files?

Comment (Trott) [reply .gitignore]: Maybe `tmp` would be a better name? We may want to have unit tests one day....

Comment (nschonni) [reply .gitignore]: Makes sense, rebased

Comment (nschonni) [reply .gitignore]: Oops, missed the `tmp` comment. I'll swap that and rebase again

Comment (nschonni) [reply .gitignore]: OK, swapped the `test` to `tmp`

**Comment by nschonni** [comment]
This could probably land before the other one, but it will just be red till the other PR.

PS: Should I add the Travis badge to the readme?

**Comment by Trott** [comment]
> PS: Should I add the Travis badge to the readme?

That would be 👍 by me.

**Comment by nschonni** [comment]
Done

**Comment by nschonni** [comment]
@Trott I think this should be good to land now that the upstream PR has landed.
Do you have the ability to turn on the Travis hook after?

**Comment by Trott** [comment]
OK, it's enabled.

**Comment by XhmikosR** [comment]
@nschonni I was looking at this and I don't understand, why is the tmp folder added in git?

**Comment by XhmikosR** [comment]
It seems like someone tried to add a submodule but then changed their mind.

File: .gitignore (hunk 1 above)
Comment (nschonni) [reply .gitignore]: @XhmikosR this is why it changed, but the rebase left the old empty folder refernce

Comment (XhmikosR) [reply .gitignore]: OK, I'm gonna remove it in a separate PR.

Comment (nschonni) [reply .gitignore]: Thanks!

**Pull request merged**